File: app/assets/javascripts/channels/status_code.js
$(function() {
  if ($('.ws_sender').attr('id') == "smoke-status-code") {
    App.status_code = App.cable.subscriptions.create("StatusCodeChannel", {
      received: function() {
        function_handler_start("Run Smoke test Status code page")
        function_handler_done("Done Smoke test Status code page")

        function_stop_stop("KILL PROCCESS TEST")
        
    }
  });

    $('.pid_process').click(function(){
      $('.stop').click()
      App.status_code.send(JSON.stringify({
        pid_cucumber: "pid_cucumber"
      }))
    });

};

    $('#smoke-status-code').click(function(){

      $(".step_cucumber").click();
      $(".start").click();
      window.scrollTo(0, 999999);

      App.status_code.send(JSON.stringify({
        smoke_status_code: true,  //!!!!!
        count_features: get_data(),
        thread: $('#generator_thread').val(),
        clear_cache: $('#generator_clear_clear_cache').prop('checked'),
        geo: $('#generator_geo').val(),
        proxy_server: $('#generator_proxy_server').val(),
        proxy_http: "--proxy-server=" + $('#generator_proxy_http').val(),
        response_http: $('#generator_display_return_only_response__200').prop('checked'),
        not_response_http: $('#generator_display_return_only_response__400').prop('checked'),
        param_headless: $('#generator_param_headless').val(),
      }));

      return false;
  })
});

File: app/channels/ws/binomo/status_code.rb
require "./features/helpers/decor"

module Server
  class StatusCode
    attr_accessor :smoke_status_code, :count_features,
                  :thread, :clear_cache, :geo, :proxy_http,
                  :proxy_server, :response_http,
                  :not_response_http, :param_headless

    def initialize(params)
      @params = params
      params.each { |key, value| public_send("#{key}=", value) }
    end

    def get_link
      JSON.parse(@params.to_json).map { |key, value| key == "count_features" ? value.to_s : "#{key}=#{value.to_s}" }.join(" ")
    end

    def message
      thread.to_i.times do
        if clear_cache == "true"
          puts "Clear cache".yellow
          system "rake tmp:cache:clear"
          system "rm -rf tmp"
          puts "Cache clear done".yellow
        end

        puts get_link.yellow
        puts;
        send_broadcast "smoke_binomo", "Run Smoke test Status code page"
        system get_link
        send_broadcast "smoke_binomo", "Done Smoke test Status code page"
      end

    end
  end
end
